### PR TITLE
Feature/unit selection modal

### PR DESCRIPTION
**Pull request by Bluesmile82**
This PR branches from `feature/indicator-selection-modal`
- Finish the dynamic selection of props and dispatch for the modal and 
- Add the new ToolModalButton to the nav

image
image

### CODE REVIEW

File: frontend/scripts/react-components/tool/tool-modal/tool-modal.component.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import BaseModal from 'react-components/tool/tool-modal/base-modal';
+import SimpleModal from 'react-components/shared/simple-modal/simple-modal.component';
+import 'react-components/tool/tool-modal/tool-modal.scss';
+
+export default function ToolModal({ items, selectedItem, onChange, activeModal, setActiveModal }) {
+  return activeModal ? (
```
Comment (sorodrigo) [inline frontend/scripts/react-components/tool/tool-modal/tool-modal.component.jsx]: why not pass this in the `isOpen` prop?

Comment (Bluesmile82) [reply frontend/scripts/react-components/tool/tool-modal/tool-modal.component.jsx]: We need the activeModal to render the BaseModal and its easier not to render the modal. Do you think they'll be some inconveniences?

Comment (sorodrigo) [reply frontend/scripts/react-components/tool/tool-modal/tool-modal.component.jsx]: ok

**Comment by Bluesmile82** [comment]
- Added special case for disabled button

Try: http://localhost:8081/flows?selectedContextId=5

image

**Pull request merged**